website/django/data/utilities.py

```diff
@@ -24,10 +28,104 @@ def update_autocomplete_words():
             """)
 
 
+def update_materialized_view():
+    with connection.cursor() as cursor:
+        cursor.execute(
+            """
+            DROP MATERIALIZED VIEW IF EXISTS currentvariant;
+                CREATE MATERIALIZED VIEW currentvariant AS (
+                    SELECT * FROM "variant" WHERE (
+                        "id" IN ( SELECT DISTINCT ON ("Genomic_Coordinate_hg38") "id" FROM "variant" ORDER BY "Genomic_Coordinate_hg38" ASC, "Data_Release_id" DESC )
+                    )
+                );
+            """
+        )
+
+
 def set_release_name_defaults(apps, schema_editor):
     count = 1
     releases = DataRelease.objects.all().order_by('date')
     for release in releases:
         release.name = count
         release.save()
         count += 1
+
+
+def send_mupit_request(query_url, params, retries=5):
+    if retries <= 0:
+        print "Request failed 5 times, exiting."
+        sys.exit(1)
+    try:
+        r = requests.post(query_url, data=params)
+        return r
+    except requests.exceptions.RequestException as e:
+        print e
+        time.sleep(10)
+        retries -= 1
+        r = send_mupit_request(query_url, params, retries)
+    return r
+
+
+def isMissenseSubstitution(ref, alt):
+    bases = ['a', 'c', 't', 'g']
+    ref = ref.lower()
+    alt = alt.lower()
+    if ref in bases and alt in bases and len(ref) == 1 and len(alt) == 1:
+        return True
+    return False
+
+
+def update_mupit_structure_for_existing_variants():
```

Comment:
Is this a one time thing, or would we need to repeat it at every release? For the first case it would be could to clearly mark this, e.g. by putting all the relevant code into a separate file.